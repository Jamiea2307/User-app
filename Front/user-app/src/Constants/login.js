export const loginData = {
  title: "Login",
  email: "Email:",
  password: "Password:",
  linkText: "Register",
};
