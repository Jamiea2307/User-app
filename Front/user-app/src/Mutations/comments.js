import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;

export const CREATE_REPLY = gql`
  mutation createReply(
    $parentPost: String!
    $parentComment: String!
    $body: String!
  ) {
    createReply(
      parentPost: $parentPost
      parentComment: $parentComment
      body: $body
    )
  }
`;
