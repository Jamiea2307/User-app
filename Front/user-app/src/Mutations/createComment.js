import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;
