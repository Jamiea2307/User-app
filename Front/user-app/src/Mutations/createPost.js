import { gql } from "@apollo/client";

export const CREATE_POST = gql`
  mutation createPost($content: String!) {
    createPost(content: $content) {
      content
    }
  }
`;
