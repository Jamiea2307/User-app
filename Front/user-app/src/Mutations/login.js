import { gql } from "@apollo/client";

export const LOGIN_USERS = gql`
  mutation loginUser($email: String!, $password: String!) {
    loginUser(email: $email, password: $password) {
      email
      password
    }
  }
`;
