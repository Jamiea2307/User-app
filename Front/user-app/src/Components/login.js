import { useState } from "react";
import { useMutation } from "@apollo/client";
import { UserFormBox, UserBox, SubmitBox } from "../Styles/userFormStyles";
import { LOGIN_USERS } from "../Mutations/login";
import { loginData } from "../Constants/login";
import { Link } from "react-router-dom";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [loginUser] = useMutation(LOGIN_USERS);

  const submitUser = async () => {
    try {
      await loginUser({
        variables: {
          email: email,
          password: password,
        },
      });
      setErrorMessage();
    } catch (err) {
      setErrorMessage(err.message);
    }
  };
  return (
    <UserFormBox>
      <h2>{loginData.title}</h2>
      <form
        className="login-form"
        onSubmit={(e) => {
          e.preventDefault();
          submitUser();
          // getUser;
        }}
      >
        <UserBox className="user-box">
          <input
            className="login-input"
            type="text"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <label>{loginData.email} </label>
        </UserBox>
        <UserBox>
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <label>{loginData.password}</label>
        </UserBox>
        <div className="errorMessage">{errorMessage}</div>
        <SubmitBox type="submit" value="Submit" />
      </form>
      <Link className="pageLink" to="/Register">
        {loginData.linkText}
      </Link>
    </UserFormBox>
  );
};

export default Login;
