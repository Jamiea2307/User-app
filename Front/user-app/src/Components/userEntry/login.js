import { useState } from "react";
import { useMutation } from "@apollo/client";
import { UserFormBox, SubmitBox } from "../../Styles/userFormStyles";
import { LOGIN_USERS } from "../../Mutations/login";
import { loginData } from "../../Constants/userContent";
import { Link } from "react-router-dom";
import { useHistory } from "react-router-dom";
import UserInput from "../Widgets/userInput";

const Login = () => {
  const history = useHistory();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [loginUser] = useMutation(LOGIN_USERS);

  const submitUser = async () => {
    try {
      await loginUser({
        variables: {
          email: email,
          password: password,
        },
      });
      history.push("/home");
    } catch (err) {
      setErrorMessage(err.message);
    }
  };
  return (
    <UserFormBox>
      <h2>{loginData.title}</h2>
      <form
        className="login-form"
        onSubmit={(e) => {
          e.preventDefault();
          submitUser();
        }}
      >
        <UserInput
          loginData={loginData.email}
          setValue={setEmail}
          type="text"
        />
        <UserInput
          loginData={loginData.password}
          setValue={setPassword}
          type="password"
        />
        <div className="errorMessage">{errorMessage}</div>
        <SubmitBox type="submit" value="Submit" />
      </form>
      <Link className="pageLink" to="/Register">
        {loginData.linkText}
      </Link>
    </UserFormBox>
  );
};

export default Login;
