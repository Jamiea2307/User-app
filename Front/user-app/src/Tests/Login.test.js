import React from "react";
import renderer from "react-test-renderer";
import Login from "../Components/userEntry/login";
import { BrowserRouter as Router } from "react-router-dom";
import { LOGIN_USERS } from "../Mutations/login";
import { MockedProvider } from "@apollo/client/testing";

describe("Login", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[]}>
          <Router>
            <Login />
          </Router>
        </MockedProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
