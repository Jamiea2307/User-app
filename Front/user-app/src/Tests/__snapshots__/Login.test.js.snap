// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="sc-bdfBwQ lozkLg"
>
  <h2>
    Login
  </h2>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="sc-dlfnbm jqeTBD user-box"
    >
      <input
        onChange={[Function]}
        type="text"
      />
      <label>
        Email:
         
      </label>
    </div>
    <div
      className="sc-dlfnbm jqeTBD user-box"
    >
      <input
        onChange={[Function]}
        type="password"
      />
      <label>
        Password:
         
      </label>
    </div>
    <div
      className="errorMessage"
    >
      
    </div>
    <input
      className="sc-gsTCUz cKOPeO"
      type="submit"
      value="Submit"
    />
  </form>
  <a
    className="pageLink"
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
